miso-web/src/main/webapp/scripts/list_assaytest.js

```diff
@@ -53,12 +53,20 @@ ListTarget.assaytest = (function () {
           sTitle: "Tissue Type",
           mData: "tissueTypeId",
           mRender: function (data, type, full) {
-            var tissueType = Utils.array.findUniqueOrThrow(
-              Utils.array.idPredicate(data),
-              Constants.tissueTypes
-            );
-            var text = (full.negateTissueType ? "NOT " : "") + tissueType.alias;
-            return '<span title="' + tissueType.description + '">' + text + "</span>";
+            if (!data) {
+              return null;
+            } else {
+              var tissueType = Utils.array.findUniqueOrThrow(
+                Utils.array.idPredicate(data),
+                Constants.tissueTypes
+              );
+              var text = (full.negateTissueType ? "NOT " : "") + tissueType.alias;
+              if (type === "display") {
+                return '<span title="' + tissueType.description + '">' + text + "</span>";
+              } else {
```

Comment:
what is the else for? Can type be filter in this case?

Reply:
`type` for `mRender` can always be "display", "sort", "filter", or "type". This is returning the same text for all, so sorting and filtering will be based on the same text that's being displayed, and the table will be able to figure out the data type (string). When `type === "display"`, there is HTML added.

We wouldn't want to include the HTML for the other `type`s, because it would cause strange behaviour with sorting and filtering

Reply:
Makes sense. Thanks!